### PR TITLE
Add input file for head-on BNS initial data

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Generates this matter & gravity configuration:

Bildschirm­foto 2023-02-01 um 10 53 21 PM

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/InputFiles/Xcts/HeadOnBns.yaml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+ResourceInfo:
+  AvoidGlobalProc0: false
+  Singletons: Auto
+
+Background: &background
+  Binary:
+    XCoords: [&x_left -20., &x_right 20.]
+    ObjectA: &star_right
+      TovStar:
+        CentralDensity: 0.0008087415253997405  # h = 1.2
```
Comment (nilsdeppe) [inline tests/InputFiles/Xcts/HeadOnBns.yaml]: Where does this and the K come from? In single stars K=100 is standard and gives a maximum mass of the NS that's ~2.2 M_sun. Also, can you use scientific notation here? I find it rather difficult to parse all the leading zeroes. 8.08e-4 is much easier to read. Also, having the central density in terms of g/cm^3 in the comment would be more helpful than the specific enthalpy.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
You can squash

File: tests/InputFiles/Xcts/HeadOnBns.yaml (hunk 1 above)
Comment (nilsvu) [reply tests/InputFiles/Xcts/HeadOnBns.yaml]: Added the reference where these numbers are from, and the other things. Thanks for your review!

**Comment by nilsdeppe** [comment]
RIP, all CI failed...

File: tests/InputFiles/Xcts/HeadOnBns.yaml [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  AvoidGlobalProc0: false
+  Singletons: Auto
+
+Background: &background
+  Binary:
+    XCoords: [&x_left -20., &x_right 20.]
+    ObjectA: &star_right
+      TovStar:
+        # Values taken from Sec. V.C and Fig. 8 in
+        # https://arxiv.org/abs/1708.07358.
```
Comment (nilsdeppe) [inline tests/InputFiles/Xcts/HeadOnBns.yaml:18]: I think we should use the numbers from https://arxiv.org/pdf/2010.14518.pdf instead. It's more recent and also evolves the binaries, and allows for a simple thermal component.

**Comment by nilsvu** [comment]
Conflict with #4684 that was merged earlier

File: tests/InputFiles/Xcts/HeadOnBns.yaml (hunk 2 above)
Comment (nilsvu) [reply tests/InputFiles/Xcts/HeadOnBns.yaml:18]: You can replace this with an EOS of your choice when generating ID :) happy to put another one here as the example. The paper you link doesn't
list the central density though, or maybe I've missed it. Also, is the thermal component implemented?

Comment (nilsdeppe) [reply tests/InputFiles/Xcts/HeadOnBns.yaml:18]: Okay, let's just leave it as-is. Like you said, we can adjust later :)

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**